plugins/in_calyptia_fleet/in_calyptia_fleet.c

```diff
@@ -468,12 +468,19 @@ static void *do_reload(void *data)
     }
     reload->flb->config->conf_path_file = reload->cfg_path;
 
-    flb_free(reload);
     sleep(5);
 #ifndef FLB_SYSTEM_WINDOWS
+    flb_free(reload);
     kill(getpid(), SIGHUP);
 #else
-    GenerateConsoleCtrlEvent(1 /* CTRL_BREAK_EVENT_1 */, 0);
+    /* using the refactor that placed  `bin_restarting` inside 
+     * `flb_config` to use it as a messaging mechanism instead of
+     * GenerateConsoleCtrlEvent to overcome the fact that windows
+     * services do not have a console and therefore cannot
+     * react to console events or handle them (fixes sc-112185).
+     */
+    reload->flb->config->bin_restarting = FLB_RELOAD_IN_PROGRESS;
```

Comment:
Is there another way this can be done without a plugin accessing a direct flag/variable from the binary ?

Reply:
fluent-bit.c could expose a function for this purpose

Reply:
or the engine..

Reply:
One of the biggest problems with that approach is that the reload needs to occur within the outer loop inside the main function but the engine runs its own separate event loop internally where those events will be captured.